pkg/scheduler/queue/user_queues.go

```diff
@@ -236,9 +238,6 @@ func (q *queues) getNextQueueForQuerier(lastUserIndex int, querierID string) (us
 			continue
 		}
 
-		q.userQueuesMx.RLock()
-		defer q.userQueuesMx.RUnlock()
```

Comment:
actually this shouldn't have been in a for loop, as the defer doesn't get called until the function returns, not when each loop iteration is done.